billing/src/awsbill.js

```diff
@@ -1,136 +1,252 @@
+        // Convert single text element to h4 in mattermost.
+        output.push('#### ' + element.text.text.replace(/\*/g, '**'));
+      }
+      break;
+    }
+    case 'divider': {
+      output.push('***');
+      break;
+    }
+  }
+
+  return output.join(' ');
+};
+
+/**
+ * @description Shows your AWS bill
  * @param {ParamsType} params list of command parameters
  * @param {?string} commandText slack text message
  * @param {!object} [secrets = {}] list of secrets
  * @return {Promise<SlackBodyType>} Slack response body
  */
 async function _command(params, commandText, secrets = {}) {
   const {
-    month_year
-  } = params;
-
-  AWS = require('aws-sdk');
-  
-  if (!secrets.awsCostExplorerAccessKeyId || !secrets.awsCostExplorerSecretAccessKey || !secrets.awsCostExplorerRegion) {
-    return { response_type: 'ephemeral', text: "You must create secrets for awsCostExplorerAccessKeyId, awsCostExplorerSecretAccessKeyId " +
-      "and awsCostExporerRegion to use this command " };
+    awsCostExplorerAccessKeyId,
+    awsCostExplorerSecretAccessKey,
+    awsCostExplorerRegion
+  } = secrets;
+
+  if (
+    !awsCostExplorerAccessKeyId ||
+    !awsCostExplorerSecretAccessKey ||
+    !awsCostExplorerRegion
+  ) {
+    return {
+      response_type: 'ephemeral', // eslint-disable-line camelcase
+      text: `You must create secrets for \`awsCostExplorerAccessKeyId\`, \`awsCostExplorerSecretAccessKey\` and \`awsCostExporerRegion\` to use this command`
+    };
   }
 
-  let now = new Date();
+  const {month_year: monthYear, __client_headers: clientHeaders} = params;
+  const getClient = () => {
+    if (clientHeaders['user-agent'].includes('Slackbot')) {
+      return 'slack';
+    }
+
+    return 'mattermost';
+  };
+
+  const client = getClient();
+
+  const AWS = require('aws-sdk');
+
+  const now = new Date();
   let month = now.getUTCMonth();
   let year = now.getUTCFullYear();
-  if (month_year != null) {
-    let arr = month_year.split('/');
+
+  if (monthYear != null) {
+    const arr = monthYear.split('/');
     if (arr.length != 2) {
-      return { response_type: 'ephemeral', text: 'Invalid month/year. Example: 11/2019 for November 2019' };
+      return {
+        response_type: 'ephemeral',
+        text: 'Invalid month/year. Example: 11/2019 for November 2019'
+      };
     }
     month = parseInt(arr[0]) - 1;
     year = parseInt(arr[1]);
+    if (month < 1 || month > 12 || year < 1900 || year > 2800) {
+      return {
+        response_type: 'ephemeral', // eslint-disable-line camelcase
+        text: 'Month or year out of range. Example: 11/2019 for November 2019'
+      };
+    }
   }
-  // determine billing period start/end. toISOString() is UTC (GMT) time, which is what AWS bills in
-  let firstOfThisMonth = new Date(year, month, 1);
-  let firstOfNextMonth;
-  if (month != 12) {
-    firstOfNextMonth = new Date(year, month + 1, 1);
-  } else {
-    firstOfNextMonth = new Date(year + 1, 1, 1);
-  }
```

Comment:
Year gets incremented when the value of `month` is greater than 12. So we don't have to condition check it.